policies/codingstyle.txt

```diff
@@ -311,24 +311,27 @@ For that case, the pattern should also include:
     rettype TYPE_do_something(TYPE_CTX *,[args])
 
 Where present the TYPE (or TYPE_CTX) argument should always be first as shown
-in the examples above. In constructors or object factory type functions (such as
-a "fetch" function), the key argument that identifies the thing be constructed
-should be placed first (if such an argument exists). For example in a "fetch"
-function the name of the algorithm being fetched would be first.
-
-The remaining argument list should be put into order of importance. Deciding the
-relative importance of arguments is necessarily somewhat subjective, but the
-following guidelines should be considered:
-
-1) Any additional "context" arguments should be first. If multiple contexts are
-   necessary then they should be in order of most general to most specific. In
-   some cases (typically with an OSSL_LIB_CTX), a context may be NULL to indicate
-   a default context should be used.
-2) Required arguments should be placed after the additional contexts.
-3) Optional arguments (i.e. those that are only be used in certain circumstances
-   and therefore may be "NULL" or empty) are placed after required arguments.
-4) Output parameters are placed after optional arguments.
-5) Callbacks are placed after output parameters.
+in the examples above. The remaining argument list should be put into order of
+importance. Deciding the relative importance of arguments is necessarily
+somewhat subjective, but the following guideline ordering of the arguments
+should be considered:
+
+1) Additional context arguments
+
+   If multiple contexts are necessary then they should be in order of most
+   general to most specific. In some cases (typically with an OSSL_LIB_CTX), a
```

Comment:
nit: maybe it would be better to move the lonely `a` to the beginning of the next line.

Reply:
What is an OSSL_LIB_CTX? We have OPENSSL_CTX.
This does reflect the confusion about naming in the function names and the typedef.

Reply:
#12621 renames OPENSSL_CTX to OSSL_LIB_CTX so this is just anticipating that change

Reply:
> nit: maybe it would be better to move the lonely a to the beginning of the next line.

I'm not feeling a strong need for this.